components/knack/sources/new-record/new-record.mjs

```diff
@@ -0,0 +1,37 @@
+import knack from "../../knack.app.mjs";
+
+export default {
+  key: "knack-new-record",
+  name: "New Record",
+  description: "Emit new event for each record created",
+  type: "source",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    knack,
+    objectKey: {
+      type: "string",
+      label: "Object Key",
+      description: "The key of the object which this record belongs to. See [the Knack API docs](https://docs.knack.com/docs/object-based-requests) for more information.",
+    },
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+  },
+  async run({ $ }) {
+    const records = await this.knack.getAllRecords($, {
+      objectKey: this.objectKey,
+    }, {}, true);
```

Comment:
I'm prefered just get the first page to don't spam the event.

Imagine with have 10k records, I thing is not the best idea emit all this records, because this I just return the first page.

Makes sense?

Reply:
Can you take a look again?